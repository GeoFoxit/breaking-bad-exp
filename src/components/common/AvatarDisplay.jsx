import { Avatar } from "@mui/material";
import React from "react";

const AvatarDisplay = ({
  imgSrc,
  alt,
}) => {
  return (
    <Avatar
      sx={{
        bgcolor: "primary.silver",
        height:"500px",
        width:"300px",
        transition: "0.2s ease",
        ":hover": { transform: "scale(1.1)" }
      }}
      variant="square"
      src={imgSrc}
    >
      {alt}
    </Avatar>
  );
};

export default React.memo(AvatarDisplay);