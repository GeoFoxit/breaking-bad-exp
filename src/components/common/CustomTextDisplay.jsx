import { Typography } from "@mui/material";
import React from "react";

const CustomTextDisplay = ({
  title,
  value,
}) => {
  return (
    <div>
      <Typography
        variant="body1"
        color="grey.500"
      >
        {title}
      </Typography>
      <Typography
        variant="h6"
      >
        {value}
      </Typography>
    </div>
  );
};

export default React.memo(CustomTextDisplay);