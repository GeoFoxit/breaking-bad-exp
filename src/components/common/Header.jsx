import React from "react";
import { AppBar, Box, Toolbar, Typography } from "@mui/material";
import StyledLink from "./StyledLink";

const Header = () => {
  return (
    <AppBar
      position="sticky"
      elevation={1}
    >
      <Toolbar
        sx={{
          p: 3,
          justifyContent: "center",
          alignItems: "center",
          flexWrap: "wrap",
          gap: "64px"
        }}
      >
        <Typography
          color="primary.main"
          variant="h5"
          align="center"
          fontWeight="bold"
          sx={{ userSelect: "none" }}
        >
            B
          <Typography
            color="white"
            variant="h5"
            align="center"
            component="span"
            fontWeight="inherit"
          >
            reaking
          </Typography>
            B
          <Typography
            color="white"
            variant="h5"
            align="center"
            component="span"
            fontWeight="inherit"
          >
            ad
          </Typography>
            E
          <Typography
            color="white"
            variant="h5"
            align="center"
            component="span"
            fontWeight="inherit"
          >
            xplorer
          </Typography>
        </Typography>
        <Box>
          <StyledLink
            to="/"
          >
            Home
          </StyledLink>
        </Box>
      </Toolbar>
    </AppBar>
  );
};

export default Header;