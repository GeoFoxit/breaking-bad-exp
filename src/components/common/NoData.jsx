import { Box, Typography } from "@mui/material";
import React from "react";

const NoData = () => {
  return (
    <Box
      width="100%"
      my={12}
    >
      <Typography
        align="center"
        variant="h6"
      >
        Sorry, no data available
      </Typography>
    </Box>
  );
};

export default React.memo(NoData);