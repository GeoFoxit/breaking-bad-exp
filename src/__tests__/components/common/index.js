import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import NoData from "../../../components/common/NoData";
import Header from "../../../components/common/Header";
import CustomTextDisplay from "../../../components/common/CustomTextDisplay";
import AvatarDisplay from "../../../components/common/AvatarDisplay";

it("NoDataComponent shows text", () => {
  // Prepare
  render(<NoData />);
  // Assert
  expect(screen.queryByText("Sorry, no data available")).toBeInTheDocument();
});

it("CustomTextDisplayComponent shows valid text", () => {
  // Prepare
  render(
    <CustomTextDisplay
      title="Test title"
      value="Test value"
    />
  );
  // Assert
  expect(screen.queryByText("Test title")).toBeInTheDocument();
  expect(screen.queryByText("Test value")).toBeInTheDocument();
});

describe("HeaderComponent", () => {
  it("Shows valid Header", () => {
    // Prepare
    const header = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    // Assert
    expect(header).toMatchSnapshot();
  });

  it("Contains a Home link", () => {
    // Prepare
    const header = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    // Assert
    expect(header
      .getByText("Home")
      .closest("a")
    ).toHaveAttribute("href", "/");
  });
});

describe("AvatarDisplayComponent", () => {
  it("Shows alt text if no src is provided", () => {
    // Prepare
    render(<AvatarDisplay alt="No Data" />);
    // Assert
    expect(screen.queryByText("No Data")).toBeInTheDocument();
    expect(screen.queryByRole("img")).not.toBeInTheDocument();
  });

  it("Shows img if valid imgSrc prop is provided", () => {
    // Prepare
    const validTestImgSrc = "https://d33wubrfki0l68.cloudfront.net/0834d0215db51e91525a25acf97433051f280f2f/c30f5/img/redux.svg";
    render(
      <AvatarDisplay
        imgSrc={validTestImgSrc}
        alt="No Data"
      />
    );
    // Assert
    expect(screen.queryByText("No Data")).not.toBeInTheDocument();
    expect(screen.queryByRole("img")).toBeInTheDocument();
  });
});
