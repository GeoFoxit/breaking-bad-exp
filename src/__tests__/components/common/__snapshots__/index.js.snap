// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Shows valid Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-10zhepo-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-smx9im-MuiToolbar-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1pa9e9e-MuiTypography-root"
          >
            B
            <span
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-nl7ql3-MuiTypography-root"
            >
              reaking
            </span>
            B
            <span
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-nl7ql3-MuiTypography-root"
            >
              ad
            </span>
            E
            <span
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-nl7ql3-MuiTypography-root"
            >
              xplorer
            </span>
          </h5>
          <div
            class="MuiBox-root css-0"
          >
            <a
              class="css-1wx78ax"
              href="/"
            >
              Home
            </a>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-10zhepo-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-smx9im-MuiToolbar-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1pa9e9e-MuiTypography-root"
        >
          B
          <span
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-nl7ql3-MuiTypography-root"
          >
            reaking
          </span>
          B
          <span
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-nl7ql3-MuiTypography-root"
          >
            ad
          </span>
          E
          <span
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-nl7ql3-MuiTypography-root"
          >
            xplorer
          </span>
        </h5>
        <div
          class="MuiBox-root css-0"
        >
          <a
            class="css-1wx78ax"
            href="/"
          >
            Home
          </a>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
